src/dbus_api/api/manager_3_8/api.rs

```diff
@@ -38,21 +38,23 @@ pub fn create_pool_method(f: &Factory<MTSync<TData>, TData>) -> Method<MTSync<TD
     f.method("CreatePool", (), create_pool)
         .in_arg(("name", "s"))
         .in_arg(("devices", "as"))
-        // Optional key description of key in the kernel keyring
-        // b: true if the pool should be encrypted and able to be
-        // unlocked with a passphrase associated with this key description.
+        // Optional key descriptions of key in the kernel keyring
+        // a: array of zero or more elements
+        // b: true if a token slot is specified
+        // i: token slot
         // s: key description
         //
-        // Rust representation: (bool, String)
-        .in_arg(("key_desc", "(bs)"))
-        // Optional Clevis information for binding on initialization.
-        // b: true if the pool should be encrypted and able to be unlocked
-        // using Clevis.
+        // Rust representation: Vec<((bool, u32), String)>
+        .in_arg(("key_desc", "a((bu)s)"))
+        // Optional Clevis infos for binding on initialization.
+        // a: array of zero or more elements
+        // b: true if a token slot is specified
+        // i: token slot
         // s: pin name
         // s: JSON config for Clevis use
         //
-        // Rust representation: (bool, (String, String))
-        .in_arg(("clevis_info", "(b(ss))"))
+        // Rust representation: Vec<((bool, u32), String, String)>
+        .in_arg(("clevis_info", "a((bu)ss)"))
```

Comment:
^```(bu)ss```^```(bu)(ss)``` OR you can modify EncryptionInfos struct in dbus_api/api/manager_3_8/methods.rs to match this spec as it is.

This is an example of pesky bug we've run into before, where the D-Bus API silently drops badly formed data if it is marshalled into an array due to a dbus-rs bug.